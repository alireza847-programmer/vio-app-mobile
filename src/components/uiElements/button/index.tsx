import React from 'react';
import Button from './style';
import {VButtonProps} from 'types/components/uiElements/button';
import VText from '../text';
import {theme} from 'themes/emotion';
import VRow from '../row';

const VButton = (props: VButtonProps) => {
  const {
    mode = 'primary',
    subTitle,
    title,
    styled = 'FULL_WIDTH',
    icon,
    rightIcon,
    onPress,
    testID,
    disabled = false,
    width,
  } = props;

  const renderContentColor = () => {
    if (disabled) {
      return theme.colors.textColor.button.disabled;
    }
    if (styled === 'TEXT') {
      return theme.colors.button[mode];
    }
    return theme.colors.textColor.button[mode];
  };

  return (
    <Button
      width={width}
      disabled={disabled}
      testID={testID}
      activeOpacity={0.8}
      onPress={onPress}
      styled={styled}
      mode={mode}>
      {icon && (
        <VRow {...(title && {marginRighRatio: 2})} fullWidth={false}>
          {icon(renderContentColor())}
        </VRow>
      )}
      <VText
        typography="semiBold16"
        color={renderContentColor()}
        marginTopRatio={0}>
        {title}
      </VText>
      {subTitle && (
        <VText
          typography="regular16"
          color={renderContentColor()}
          marginLeftRatio={1.2}
          marginTopRatio={0}>
          {subTitle}
        </VText>
      )}
      {rightIcon && (
        <VRow {...(title && {marginLeftRatio: 2})} fullWidth={false}>
          {rightIcon(renderContentColor())}
        </VRow>
      )}
    </Button>
  );
};
export default VButton;
