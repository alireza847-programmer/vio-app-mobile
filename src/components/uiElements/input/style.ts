import styled from '@emotion/native';

const Input = styled.TextInput(({theme}) => ({
  borderWidth: 1,
  width: '100%',
  padding: theme.HorizontalRem * 4,
  borderRadius: theme.radius * 2,
  borderColor: theme.colors.stroke.input,
  fontFamily: theme.fonts.medium15.fontFamily,
  fontSize: theme.fonts.medium15.fontSize,
}));

export default Input;
