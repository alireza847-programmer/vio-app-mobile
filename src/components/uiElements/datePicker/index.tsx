import {theme} from '@storybook/react-native';
import React from 'react';
import {View} from 'react-native';
import Calendar from 'react-native-calendar-range-picker';
import {VDatePickerProps} from 'types/components/uiElements/datePicker';

const VDatePicker = (props: VDatePickerProps) => {
  const {...rest} = props;
  return (
    <View style={{width: '100%'}}>
      <Calendar disabledBeforeToday pastYearRange={0} {...rest} />
    </View>
  );
};

export default VDatePicker;
