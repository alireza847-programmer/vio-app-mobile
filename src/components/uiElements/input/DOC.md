# VInput Component

## Overview

The `VInput` component is a customizable input field designed for use in React Native applications. It provides a clean and consistent interface for capturing user input.

## Style

The styling for the `VInput` component is defined in the `Input` module. It uses Emotion's styling system for enhanced readability and maintainability.

```jsx
import styled from '@emotion/native';
import {InputProps} from 'types/components/uiElements/input';

const VInput =
  styled.TextInput <
  InputProps >
  (({theme, marginTopRatio = 1}) => ({
    borderWidth: 1,
    width: '100%',
    padding: theme.HorizontalRem * 4,
    borderRadius: theme.radius * 2,
    borderColor: theme.colors.stroke.input,
    fontFamily: theme.fonts.medium15.fontFamily,
    fontSize: theme.fonts.medium15.fontSize,
    marginTop: theme.verticalRem * marginTopRatio,
  }));

export default VInput;
```

## Usage

```jsx
import React, {useState} from 'react';
import {View, Text} from 'react-native';
import VInput from './path-to-your-component/VInput';

const ExampleScreen = () => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (text: string) => {
    setInputValue(text);
  };

  return (
    <View>
      <Text>Enter your text:</Text>
      <VInput
        value={inputValue}
        onChangeText={handleInputChange}
        placeholder="Type here..."
        marginTopRatio={2}
      />
    </View>
  );
};

export default ExampleScreen;
```

## Props

### VInput Props

- `marginTopRatio` (number): The ratio to set the top margin for the input.

### Input Props

The `VInput` component accepts standard `TextInputProps` from React Native.
