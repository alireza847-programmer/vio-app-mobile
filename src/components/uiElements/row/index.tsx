import styled from '@emotion/native';
import {PropsWithChildren} from 'react';
import {VRowProps} from 'types/components/uiElements/row';

const VRow = styled.View<PropsWithChildren<VRowProps>>(
  ({
    theme,
    backgroundColor,
    fullWidth = true,
    justifyContent = 'flex-start',
    alignItems = 'center',
    marginTopRatio = 0,
    marginLeftRatio = 0,
    paddingHorizontalRatio = 0,
    paddingVerticalRatio = 0,
    paddingLeftRatio = 0,
    paddingRightRatio = 0,
    marginRighRatio = 0,
    minWidth,
    height,
    width,
  }) => ({
    flexDirection: 'row',
    justifyContent,
    alignItems,
    marginTop: theme.verticalRem * marginTopRatio,
    marginLeft: theme.HorizontalRem * marginLeftRatio,
    marginRight: theme.HorizontalRem * marginRighRatio,
    paddingHorizontal: theme.HorizontalRem * paddingHorizontalRatio,
    paddingVertical: theme.verticalRem * paddingVerticalRatio,
    ...(paddingLeftRatio && {
      paddingLeft: theme.HorizontalRem * paddingLeftRatio,
    }),
    ...(paddingRightRatio && {
      paddingRight: theme.HorizontalRem * paddingRightRatio,
    }),
    ...(backgroundColor && {backgroundColor}),
    ...(fullWidth && {width: '100%'}),
    ...(minWidth && {minWidth}),
    ...(width && {width}),
    ...(height && {height}),
  }),
);

export default VRow;
