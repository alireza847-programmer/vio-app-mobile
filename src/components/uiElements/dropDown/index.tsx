import React, {Fragment, useEffect, useState} from 'react';
import {VDropDownProps} from 'types/components/uiElements/dropDown';
import {theme} from 'themes/emotion';
import {ChevronSvg} from 'assets/svgs';
import {Container} from './style';
import WheelPicker from 'react-native-wheely';
import VModal from '../modal';
import VText from '../text';
import VRow from '../row';
import VButton from '../button';
import {TouchableOpacity} from 'react-native';

const VDropDown = (props: VDropDownProps) => {
  const {data, placeHolder, onChange, defaultValue} = props;
  const [visibleModal, setVisibleModal] = useState(false);
  const [value, setValue] = useState('');
  const [selectedIndex, setSelectedIndex] = useState(0);
  useEffect(() => {
    if (defaultValue) {
      setValue(String(defaultValue));
      setSelectedIndex(data.indexOf(String(defaultValue)));
    } else {
      setValue(String(placeHolder));
    }
  }, [placeHolder, defaultValue]);
  const onConfirm = () => {
    setVisibleModal(false);
    setValue(data[selectedIndex]);
    onChange(data[selectedIndex]);
  };
  const renderModal = () => {
    return (
      <VModal
        withClose
        withHeader
        title="Child 1 age"
        onClose={() => setVisibleModal(false)}
        isVisible={visibleModal}>
        <WheelPicker
          selectedIndex={selectedIndex}
          options={data}
          onChange={index => setSelectedIndex(index)}
          containerStyle={{width: '90%'}}
          opacityFunction={x => 0.3}
          selectedIndicatorStyle={{
            backgroundColor: theme.colors.button.secondary,
            borderTopWidth: 1,
            borderBottomWidth: 1,
            borderColor: theme.colors.button.primary,
          }}
          visibleRest={1}
          itemStyle={{
            backgroundColor: 'transparent',
          }}
          itemTextStyle={{
            ...theme.fonts.semiBold18,
          }}
        />
        <VRow paddingVerticalRatio={4} justifyContent="center">
          <VButton onPress={onConfirm} width={'90%'} title="CONFIRM" />
        </VRow>
      </VModal>
    );
  };
  return (
    <Fragment>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => setVisibleModal(true)}>
        <Container
          alignItems="center"
          justifyContent="space-between"
          fullWidth={false}>
          <VText
            width={theme.HorizontalRem * 10}
            textAlign="left"
            typography="semiBold16">
            {value}
          </VText>
          <VRow fullWidth={false} marginLeftRatio={3}>
            <ChevronSvg fill={theme.colors.button.primary} />
          </VRow>
        </Container>
      </TouchableOpacity>
      {renderModal()}
    </Fragment>
  );
};
export default VDropDown;
