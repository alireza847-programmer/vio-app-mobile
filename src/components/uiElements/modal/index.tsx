import Modal from 'react-native-modal';
import React, {PropsWithChildren} from 'react';
import {VModalProps} from 'types/components/uiElements/modal';
import {HeaderContainer, Wrapper, modalStyle} from './style';
import {theme} from 'themes/emotion';
import {CloseSvg} from 'assets/svgs';
import VText from '../text';
import VButton from '../button';

const VModal = (props: PropsWithChildren<VModalProps>) => {
  const {children, isVisible, onClose, title, withClose, withHeader} = props;
  return (
    <Modal
      animationIn="slideInUp"
      animationOut={'slideOutDown'}
      style={modalStyle}
      testID="modal"
      hasBackdrop={true}
      onBackButtonPress={onClose}
      onBackdropPress={() => onClose()}
      isVisible={isVisible}>
      <Wrapper>
        {withHeader && (
          <HeaderContainer
            justifyContent="space-between"
            paddingHorizontalRatio={4}
            paddingVerticalRatio={6}>
            {withClose && (
              <VButton
                onPress={onClose}
                testID="close-button"
                styled="TEXT"
                icon={fill => <CloseSvg fill={theme.colors.button.primary} />}
              />
            )}
            <VText typography="semiBold16">{title}</VText>
            <CloseSvg fill={'transparent'} />
          </HeaderContainer>
        )}
        {children}
      </Wrapper>
    </Modal>
  );
};
export default VModal;
