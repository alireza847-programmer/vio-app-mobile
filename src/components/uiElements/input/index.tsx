import React from 'react';
import Input from './style';
import {VInputProps} from 'types/components/uiElements/input';
import {theme} from 'themes/emotion';

const VInput = (props: VInputProps) => {
  const {...rest} = props;
  return (
    <Input
      placeholderTextColor={theme.colors.textColor.placeHolder}
      {...rest}
    />
  );
};

export default VInput;
