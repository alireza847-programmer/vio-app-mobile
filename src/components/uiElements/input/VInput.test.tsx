import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import VInput from './index';
import {it, describe, expect} from '@jest/globals';
import {VInputProps} from 'types/components/uiElements/input';
import {ThemeProvider} from '@emotion/react';
import {theme} from 'themes/emotion';

const Component = (props: VInputProps) => (
  <ThemeProvider theme={theme}>
    <VInput {...props} />
  </ThemeProvider>
);

describe('VInput component', () => {
  it('renders correctly with default props', () => {
    const {toJSON} = render(<Component />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom marginTopRatio', () => {
    const {toJSON} = render(<Component marginTopRatio={2} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
