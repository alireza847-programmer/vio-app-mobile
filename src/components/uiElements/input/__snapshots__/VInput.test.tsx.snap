// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VInput component renders correctly with custom marginTopRatio 1`] = `
<TextInput
  marginTopRatio={2}
  placeholderTextColor="#A3ACB8"
  style={
    [
      {
        "borderColor": "#BED2E9",
        "borderRadius": 15,
        "borderWidth": 1,
        "fontFamily": "Inter-Medium",
        "fontSize": 15,
        "marginTop": 13.34,
        "padding": 30,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;

exports[`VInput component renders correctly with default props 1`] = `
<TextInput
  placeholderTextColor="#A3ACB8"
  style={
    [
      {
        "borderColor": "#BED2E9",
        "borderRadius": 15,
        "borderWidth": 1,
        "fontFamily": "Inter-Medium",
        "fontSize": 15,
        "marginTop": 6.67,
        "padding": 30,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
